code/modules/transport/tram/tram_controls.dm

```diff
@@ -0,0 +1,202 @@
+
+/**
+ * Finds the tram from the console
+ *
+ * Locates tram parts in the lift global list after everything is done.
+ */
+/obj/machinery/computer/tram_controls/proc/find_tram()
+	for(var/datum/transport_controller/linear/transport as anything in SStransport.transports_by_type[TRANSPORT_TYPE_TRAM])
+		if(transport.specific_transport_id == specific_transport_id)
+			transport_ref = WEAKREF(transport)
+			return
+
+/obj/machinery/computer/tram_controls/ui_state(mob/user)
+	return GLOB.not_incapacitated_state
+
+/obj/machinery/computer/tram_controls/ui_status(mob/user,/datum/tgui/ui)
+	var/datum/transport_controller/linear/tram/tram = transport_ref?.resolve()
+
+	if(tram?.controller_active)
+		return UI_CLOSE
+	if(!in_range(user, src) && !isobserver(user))
+		return UI_CLOSE
+	return ..()
+
+/obj/machinery/computer/tram_controls/ui_interact(mob/user, datum/tgui/ui)
+	. = ..()
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "TramControl", name)
+		ui.open()
+
+/obj/machinery/computer/tram_controls/ui_data(mob/user)
+	var/datum/transport_controller/linear/tram/tram_controller = transport_ref?.resolve()
+	var/list/data = list()
+	data["moving"] = tram_controller?.controller_active
+	data["broken"] = (tram_controller ? FALSE : TRUE) || (tram_controller?.paired_cabinet ? FALSE : TRUE)
+	var/obj/effect/landmark/transport/nav_beacon/tram/platform/current_loc = tram_controller?.idle_platform
+	if(current_loc)
+		data["tram_location"] = current_loc.name
+	return data
+
+/obj/machinery/computer/tram_controls/ui_static_data(mob/user)
+	var/list/data = list()
+	data["destinations"] = get_destinations()
+	return data
+
+/**
+ * Finds the destinations for the tram console gui
+ *
+ * Pulls tram landmarks from the landmark gobal list
+ * and uses those to show the proper icons and destination
+ * names for the tram console gui.
+ */
+/obj/machinery/computer/tram_controls/proc/get_destinations()
+	. = list()
+	for(var/obj/effect/landmark/transport/nav_beacon/tram/platform/destination as anything in SStransport.nav_beacons[specific_transport_id])
+		var/list/this_destination = list()
+		this_destination["name"] = destination.name
+		this_destination["dest_icons"] = destination.tgui_icons
+		this_destination["id"] = destination.platform_code
+		. += list(this_destination)
+
+/obj/machinery/computer/tram_controls/ui_act(action, params)
+	. = ..()
+	if(.)
+		return
+
+	switch(action)
+		if("send")
+			var/obj/effect/landmark/transport/nav_beacon/tram/platform/destination_platform
+			for(var/obj/effect/landmark/transport/nav_beacon/tram/platform/destination as anything in SStransport.nav_beacons[specific_transport_id])
+				if(destination.platform_code == params["destination"])
+					destination_platform = destination
+					break
+
+			if(isnull(destination_platform))
+				return FALSE
+
+			SEND_SIGNAL(src, COMSIG_TRANSPORT_REQUEST, specific_transport_id, destination_platform.platform_code)
```

Comment:
i dont understand why this is a signal instead of a proc call

Reply:
it can be, I was treating the controls the same as tram request buttons which also send signals